tests/PHPStan/Rules/DeadCode/data/call-to-static-method-without-impure-points.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace CallToStaticMethodWithoutImpurePoints;
+
+function (): void {
+	X::myFunc();
+	X::myFUNC();
+	X::throwingFUNC();
+	X::throwingFunc();
+	X::funcWithRef();
+	X::impureFunc();
+	X::callingImpureFunc();
+
+	$a = X::myFunc();
```

Comment:
Why do you think this line should error?

Reply:
I thought it should, because we do a similar error for functions: https://github.com/phpstan/phpstan-src/commit/57b0ae06e2fa4e2a43e9deb9f27afb8b24166bc9